core-bundle/src/Resources/contao/classes/DataContainer.php

```diff
@@ -1541,6 +1541,106 @@ public static function getDriverForTable(string $table): string
 
 		return $dataContainer;
 	}
+
+	/**
+	 * Generates the label for a given data record according to the DCA configuration.
+	 *
+	 * @param array  $row   The data record
+	 * @param string $table The name of the data container
+	 */
+	public static function generateRecordLabel($row, $table): string
```

Comment:
I don't think this is a good idea, is it? It's a static method that has dependencies to other services.
But then again, this happens all over the place in the "old" DCA stuff.

Reply:
Yeah, let's leave it like that then. Every DC is instantiated using `new ...` directly anyway and does not get the DB injected. We can only fix that by completely rewriting lots of parts. Possibly based on the new config stuff then which @SeverinGloeckle seems to be working on 😉

Reply:
Can we maybe use an instance method instead? Then we can at least refactor the service usages to a locator or shared properties at some point. 

Personally, I'd go with a helper service because I think this functionality is quite orthogonal to the rest of the DC concept (maybe the `row` parameter needs to be named `fields`). Even after the data containers are gone this could IMHO live on for BC without hurting anyone. But up to you.

Reply:
> Can we maybe use an instance method instead?

The idea was that @bezin for example could use this method for his PR as well. Although he might have a DataContainer instance there as well? 🤔

Reply:
@bezin Do you have a DataContainer instance or do you need the static method?

Reply:
Actuall I think we can make it non static. A DataContainer instance is available here: https://github.com/contao/contao/pull/3498/files#diff-f2b278b76e6ee91923f9b1ebfdedf63d82cf70c4ef365467acb11b011b7642adR91

Reply:
Yes, I have one as I have to create one for the `label_callback` anyways.

Reply:
Then I'll change it accordingly.